src/main/java/src/Product.java

```diff
@@ -1,128 +1,121 @@
 package src;
 
 import java.util.ArrayList;
-import java.util.Arrays;
 
 public class Product implements java.io.Serializable {
-    private Integer quantity;
-    private String name;
-    private Double price;
-    private Boolean isPurchased;
-
-
-    public Product(int quantity, String name, Double price, Boolean isPurchased) {
-
-        this.quantity = quantity;
-        this.name = sanitizeString(name);
-        this.price = price;
-        this.isPurchased = isPurchased;
-    }
-
-    private static String sanitizeString(String name) {
-        return name.replace("\n", "").replace("%%", "").replace("}}", "");
-    }
-
-    public int getQuantity() {
-        return quantity;
-    }
-
-    public String getName() {
-        return name;
-    }
-
-    public Double getPrice() {
-        return price;
-    }
-
-    public Boolean getPurchased() {
-        return isPurchased;
+  private final Integer quantity;
+  private final String name;
+  private final Double price;
+  private final Boolean isPurchased;
+
+  public Product(int quantity, String name, Double price, Boolean isPurchased) {
+
+    this.quantity = quantity;
+    this.name = sanitizeString(name);
+    this.price = price;
+    this.isPurchased = isPurchased;
+  }
+
+  private static String sanitizeString(String name) {
+    return name.replace("\n", "").replace("%%", "").replace("}}", "");
+  }
+
+
+  public int getQuantity() {
+    return quantity;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public Double getPrice() {
+    return price;
+  }
+
+  public Boolean getPurchased() {
+    return isPurchased;
+  }
+
+  @Override
+  public String toString() {
+    return "Product{"
+        + "quantity="
+        + quantity
+        + ", name='"
+        + name
+        + '\''
+        + ", price="
+        + price
+        + ", isPurchased="
+        + isPurchased
+        + '}';
+  }
+
+  public static String serializeProduct(Product product) {
```

Comment:
I don't think this method should be static.

Reply:
The method has been fixed to not static.